src/app/qgisapp.cpp

```diff
@@ -16046,6 +16046,61 @@ void QgisApp::triggerCrashHandler()
 #endif
 }
 
+void QgisApp::addTabifiedDockWidget( Qt::DockWidgetArea area, QDockWidget *dockWidget, const QString &onTopOf )
+{
+  QList< QDockWidget * > dockWidgetsInArea;
+  const auto dockWidgets = findChildren< QDockWidget * >();
+  for ( QDockWidget *w : dockWidgets )
+  {
+    if ( w->isVisible() && dockWidgetArea( w ) == area )
+    {
+      dockWidgetsInArea << w;
+    }
+  }
+
+  addDockWidget( area, dockWidget );  // First add the dock widget, then attempt to tabify
+  if ( dockWidgetsInArea.length() > 0 )
+  {
+    QDockWidget *onTopOfDockWidget = nullptr;
+    if ( !onTopOf.isEmpty() )
+    {
+      for ( QDockWidget *cw : dockWidgetsInArea )
+      {
+        if ( cw->windowTitle() == onTopOf )
```

Comment:
This should use objectName and not the window title. Otherwise code will break in translated versions

Reply:
Done!